src/Testcontainers/Builders/EnvironmentVariableBase64Provider.cs

```diff
@@ -0,0 +1,90 @@
+﻿namespace DotNet.Testcontainers.Builders
+{
+  using System;
+  using System.Linq;
+  using System.Text;
+  using System.Text.Json;
+  using DotNet.Testcontainers.Configurations;
+  using JetBrains.Annotations;
+  using Microsoft.Extensions.Logging;
+
+  internal sealed class EnvironmentVariableBase64Provider : IDockerRegistryAuthenticationProvider
+  {
+    private readonly JsonElement rootElement;
+    private readonly ILogger logger;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="EnvironmentVariableBase64Provider"/> class.
+    /// </summary>
+    /// <param name="logger">The logger.</param>
+    [PublicAPI]
+    public EnvironmentVariableBase64Provider(ILogger logger)
+    {
+      var environmentVariableValue = Environment.GetEnvironmentVariable("DOCKER_AUTH_CONFIG");
```

Comment:
Can we move this to `CustomConfiguration`? Something like `public JsonElement GetDockerAuthConfig()`? `EnvironmentConfiguration` and `PropertiesFileConfiguration` will be singeltons to access the configuration in the future.